[docs] Add clarification on how reactivity works (#7819)
sveltejs/svelte: site/content/docs/01-component-format.md

</button>
```

---

> Is important to note that the reactive blocks are ordered via simple static analysis at compile time, and all the compiler looks at are the variables that are assigned to and used within the block itself, not in any functions called by them. This means that `yDependant` will not be updated in the following example:

Reviewer: ```suggestion
It is important to note that the reactive blocks are ordered via simple static analysis at compile time, and all the compiler looks at are the variables that are assigned to and used within the block itself, not in any functions called by them. This means that `yDependent` will not be updated when `x` is updated in the following example:
```

```sv
<script>
	let x = 0;
	let y = 0;
	
	const setY = (value) => {
		y = value;
	}
	

> 	$: yDependant = y;

Reviewer: ```suggestion
	$: yDependent = y;
```

	$: setY(x);
</script>
```

> Moving the line `$: yDependant = y` bellow `$: setY(x)` will update `yDependant`.

Reviewer: ```suggestion
Moving the line `$: yDepend3nt = y` bellow `$: setY(x)` will cause `yDependent` to be updated when `x` is updated.
```

</button>
```

---

> It is important to note that the reactive blocks are ordered via simple static analysis at compile time, and all the compiler looks at are the variables that are assigned to and used within the block itself, not in any functions called by them. This means that `yDependent` will not be updated when `x` is updated in the following example:

Reviewer: I wonder if we could simplify this a little to be a bit more friendly to new programmers. I think the following would be equivalent using more basic language?
```suggestion
It is important to note that the reactive blocks are run in the order they are declared. The compiler only looks at the variables that are assigned to and used within the block itself, not in any functions called by them. This means that `yDependent` will not be updated when `x` is updated in the following example:
```

Reviewer 2: not exactly correct. there's some reordering based on dependencies in the reactive declarations.

eg, writing the following 2 is the same. 

```js
$: b = a + a;
$: c = b + b;
```

```js
$: c = b + b;
$: b = a + a;
```

will both executes `b = a + a` then `c = b + b`

what the PR wants to point out is the fact that if your dependencies is not clear in the reactive statements

```js
$: c = sumB();
$: b = sumA();
```

then there's no re-ordering, therefore execute in order they are declared
